src/uri/uri.cc

```diff
@@ -240,13 +245,49 @@ auto URI::path() const -> std::optional<std::string> {
     return "/" + this->path_.value();
   }
 
-  size_t path_pos = this->data.find(this->path_.value());
-  if (path_pos != std::string::npos && path_pos > 0 &&
-      this->data[path_pos - 1] == '/') {
-    return "/" + this->path_.value();
+  return path_;
+}
+
+auto URI::path(const std::string &path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
   }
 
-  return path_;
+  const auto is_relative = path.starts_with(".");
+  if (is_relative) {
+    throw URIError{"You cannot set a relative path"};
+  }
+
+  auto parsed_path = URI{std::string{path}};
+  if (!parsed_path.path().has_value()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  this->path_ = parsed_path.path().value();
+  return *this;
+}
+
+auto URI::path(std::string &&path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  const auto is_relative_path = path.starts_with(".");
+  if (is_relative_path) {
+    throw URIError{"You cannot set a relative path"};
+  }
+
+  auto parsed_path = URI{std::move(path)};
+  if (!parsed_path.path().has_value()) {
```

Comment:
I think this whole conditional goes away if you do `this->path_ = parsed_path.path()`?

Reply:
yeah as it is an optional too :)